back/src/controllers/categories/GetCategories.go

```diff
@@ -30,14 +30,15 @@ func GetCategories(ctx *fiber.Ctx) error {
 func queryCategories(page int, ctx *fiber.Ctx) ([]model.Category, error) {
 	pagLimit := constants.PaginationLimit
 	offset := (page - 1) * pagLimit
+	db := config.Database
 
-	categories := make([]model.Category, pagLimit)
-	errQuery := config.Database.Limit(pagLimit).Offset(offset).Order("order_of_display asc").Find(&categories)
-	if errQuery.Error != nil {
+	cats := make([]model.Category, pagLimit)
+	err := db.Limit(pagLimit).Preload("Products").Offset(offset).Order("order_of_display asc").Find(&cats)
```

Comment:
I think it would be useful to refactor this function and make it available for multiple of our cases.

Reply:
Good idea I will try to implement it

Reply:
Let's do it in an other PR later so it doesn't block the mobile